source/upgrade.txt

```diff
@@ -0,0 +1,123 @@
+.. _php-upgrade:
+
+=======================
+Upgrade Driver Versions
+=======================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: compatibility, backwards compatibility
+
+Overview
+--------
+
+On this page, you can learn how to upgrade your driver to a new version. This
+page also includes the changes you must make to your application when you 
+upgrade to a new version of the {+php-library+}.
+
+How to Upgrade
+--------------
+
+Before you upgrade, perform the following actions:
+
+- Ensure the new {+driver-short+} version is compatible with the {+mdb-server+} versions
+  your application connects to and the PHP version your
+  application compiles with. For version compatibility information, see the
+  :ref:`{+php-library+} Compatibility <php-compatibility>`
+  page.
+- Address any breaking changes between the driver version
+  your application is using and your planned upgrade version in the
+  :ref:`Breaking Changes <php-breaking-changes>` section.
+
+.. tip::
+
+   To ensure compatibility across {+mdb-server+} versions when
+   upgrading driver versions, use the :ref:`{+stable-api+} <php-stable-api>`.
+
+To upgrade your driver version, replace ``<version-number>`` with the version number 
+you want to upgrade to and run the following command in your application's 
+directory:
+
+.. code-block:: bash
+
+   composer require mongodb/mongodb:<version-number>
+
+Installation instructions for the ``mongodb`` extension may be found in the 
+:php:`PHP.net documentation <manual/en/mongodb.installation.php>`.
+
+.. _php-breaking-changes:
+
+Breaking Changes
```

Comment:
A couple of notes here:
* We don't consider dropping support for a server version a breaking change, otherwise it would only be done in a major version. When we drop support for a server version, it has been out of official support for several years.
* Deprecations are not considered breaking changes - rather is it a way to announce an upcoming breaking change that will be made in a future major version.
* Upgrading the extension requirement or PHP requirements are also not breaking changes, as composer's dependency resolution will refuse to install a version of the library that isn't compatible with the installed extension version or the installed PHP version.

To summarise, this section should mention that we will make breaking changes only in a new major version, and we're also providing an upgrade file to go along with that version, along with deprecating any functionality to be removed. For example, we're currently working on version 2.0 of the extension, and the [corresponding upgrade file](https://github.com/mongodb/mongo-php-driver/blob/v2.x/UPGRADE-2.0.md) lists everything that is removed. Note that this is a work in progress and should not yet be added to the docs.

We can list dropping support for MongoDB 3.6 in version 1.20 as a breaking change, but the other changes listed are not breaking changes in a traditional sense and should be removed.

Reply:
I will remove all the deprecation notes other than the dropped support for MongoDB Server 3.6 for Version 1.20. 
Once version 2.0 is released, we can add a note about upgrade files, as there are no other upgrade files for current versions out (the last one I see was for upgrading to 1.5 which is now the oldest relevant version according to compatibility table). Let me know know if this looks good, thanks!